From 8f77bd63d8b5b2ab593b134fef2bf5e59bed0441 Mon Sep 17 00:00:00 2001
From: Oliver Heger <oheger@apache.org>
Date: Fri, 3 Jan 2014 19:38:03 +0000
Subject: [PATCH] [BEANUTILS-456] Provide another method for querying a write
 method.

PropertyUtilsBean.getWriteMethod(PropertyDescriptor) may fail for properties
obtained via custom bean introspection. Added a warning to the Javadocs. The
method PropertyUtilsBean.getWriteMethod(Class, PropertyDescriptor) was made
public because it works correctly in this case.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/beanutils/trunk@1555231 13f79535-47bb-0310-9956-ffa450edef68
---
 .../org/apache/commons/beanutils/PropertyUtilsBean.java   | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/main/java/org/apache/commons/beanutils/PropertyUtilsBean.java b/src/main/java/org/apache/commons/beanutils/PropertyUtilsBean.java
index 0b767306e..a55f2a7c5 100644
--- a/src/main/java/org/apache/commons/beanutils/PropertyUtilsBean.java
+++ b/src/main/java/org/apache/commons/beanutils/PropertyUtilsBean.java
@@ -1274,6 +1274,11 @@ public Object getSimpleProperty(Object bean, String name)
      * <p>Return an accessible property setter method for this property,
      * if there is one; otherwise return <code>null</code>.</p>
      *
+     * <p><em>Note:</em> This method does not work correctly with custom bean
+     * introspection under certain circumstances. It may return {@code null}
+     * even if a write method is defined for the property in question. Use
+     * {@link #getWriteMethod(Class, PropertyDescriptor)} to be sure that the
+     * correct result is returned.</p>
      * <p><strong>FIXME</strong> - Does not work with DynaBeans.</p>
      *
      * @param descriptor Property descriptor to return a setter for
@@ -1295,8 +1300,9 @@ public Method getWriteMethod(PropertyDescriptor descriptor) {
      * @param clazz The class of the read method will be invoked on
      * @param descriptor Property descriptor to return a setter for
      * @return The write method
+     * @since 1.9.1
      */
-    Method getWriteMethod(Class<?> clazz, PropertyDescriptor descriptor) {
+    public Method getWriteMethod(Class<?> clazz, PropertyDescriptor descriptor) {
         BeanIntrospectionData data = getIntrospectionData(clazz);
         return (MethodUtils.getAccessibleMethod(clazz,
                 data.getWriteMethod(clazz, descriptor)));
